ts/Series/FlowMap/FlowMapSeries.ts

```diff
@@ -0,0 +1,834 @@
+            offsetY *= offset;
+
+            toX += offsetX;
+            toY += offsetY;
+        }
+
+        // Vector between the points.
+        let dX = toX - fromX,
+            dY = toY - fromY;
+
+        // Vector is halved.
+        dX *= 0.5;
+        dY *= 0.5;
+
+        // Vector points exactly between the points.
+        let mX = fromX + dX,
+            mY = fromY + dY;
+
+        // Rotating the halfway distance by 90 anti-clockwise.
+        // We can then use this to create an arc.
+        let tmp = dX;
+        dX = dY;
+        dY = -tmp;
+
+        // Weight vector calculation for the middle of the curve.
+        let [wX, wY] = FlowMapSeries.normalize(dX, dY);
+
+        // The `fineTune` prevents an obvious mismatch along the curve.
+        const fineTune = 1 + Math.sqrt(curveFactor * curveFactor) * 0.25;
+        wX *= scaledWeight * fineTune;
+        wY *= scaledWeight * fineTune;
+
+        // Ccalculate the arc strength.
+        let arcPointX = (mX + dX * curveFactor),
+            arcPointY = (mY + dY * curveFactor);
+
+        // Calculate edge vectors in the from-point.
+        let [fromXToArc, fromYToArc] =
+            FlowMapSeries.normalize(
+                arcPointX - fromX,
+                arcPointY - fromY
+            );
+
+        tmp = fromXToArc;
+        fromXToArc = fromYToArc;
+        fromYToArc = -tmp;
+
+        fromXToArc *= scaledWeight;
+        fromYToArc *= scaledWeight;
+
+
+        // Calculate edge vectors in the to-point.
+        let [toXToArc, toYToArc] =
+            FlowMapSeries.normalize(
+                arcPointX - toX,
+                arcPointY - toY
+            );
+
+        tmp = toXToArc;
+        toXToArc = -toYToArc;
+        toYToArc = tmp;
+
+        toXToArc *= scaledWeight;
+        toYToArc *= scaledWeight;
+
+        // Shrink the starting edge and middle thickness to make it grow
+        // towards the end.
+        if (growTowards) {
+            fromXToArc /= scaledWeight;
+            fromYToArc /= scaledWeight;
+            wX /= 4;
+            wY /= 4;
+        }
+
+        const shapeArgs: SVGAttributes = {
+            d: [
+                [
+                    'M',
+                    fromX - fromXToArc,
+                    fromY - fromYToArc
+                ],
+                [
+                    'Q',
+                    arcPointX - wX,
+                    arcPointY - wY,
+                    toX - toXToArc,
+                    toY - toYToArc
+                ],
+                // This is where markerEnd will be spliced to.
+                [
+                    'L',
+                    toX + toXToArc,
+                    toY + toYToArc
+                ],
+                [
+                    'Q',
+                    arcPointX + wX,
+                    arcPointY + wY,
+                    fromX + fromXToArc,
+                    fromY + fromYToArc
+                ],
+                ['Z']
+            ]
+        };
```

Comment:
Can we use formatting like this: `],[` instead of the current one? We could save a bit space, and the code might get more readable.

Reply:
I agree with your suggestion, but personallly I think it's more readable for me. `SankeySeries` also has the same formatting. Maybe we should decide and have a lint rule about this?

Reply:
That's a good idea, I think we should have a general rule for that (and for objects, if we already don't have one?).

Reply:
I've been doing some serious and deep thinking about this. In order to resolve this, I think the `], [` should be honored